src/waveform/renderers/waveformmark.h

```diff
@@ -1,39 +1,35 @@
 #ifndef WAVEFORMMARK_H
 #define WAVEFORMMARK_H
 
-#include <QString>
+#include <QDomNode>
 #include <QImage>
-#include <QColor>
 
-#include "preferences/usersettings.h"
-#include "skin/skincontext.h"
+#include "util/memory.h"
+
+#include "waveform/renderers/waveformmarkproperties.h"
 
 class ControlProxy;
-class QDomNode;
+class SkinContext;
 class WaveformSignalColors;
 
 class WaveformMark {
   public:
     WaveformMark();
-    ~WaveformMark();
+
     void setup(const QString& group, const QDomNode& node,
                const SkinContext& context,
                const WaveformSignalColors& signalColors);
-    void setKeyAndIndex(const ConfigKey& key, int i);
 
-  private:
-    ControlProxy* m_pPointCos;
+    std::unique_ptr<ControlProxy> m_pPointCos;
 
-    QColor m_color;
-    QColor m_textColor;
-    QString m_text;
-    Qt::Alignment m_align;
-    QString m_pixmapPath;
+    const WaveformMarkProperties& properties() const;
```

Comment:
I think we are using explicit getters in most classes, i.e. getProperties()? Not sure, maybe the coding style guide has an answer.